src/components/access-type.js

```diff
@@ -195,11 +198,14 @@ class AccessTypeBase extends React.Component {
                 campaignSubscriptionGroups
               );
             });
+            callback();
           }
         );
       } catch (e) {
         console.log(`Subscription / payments failed`, e);
       }
+    } else {
+      callback();
     }
```

Comment:
callback the function so that these functions will call only after all the calls are done at client side.

Reply:
why is calling `callback` needed if the `user` is not present? I think it will cause the same error, because step 2 (i.e. setUser) was not successful.

Reply:
This callback can be anything. In our case, this callback to change the view based on the access value. So if I didn't call the `callback` here, the view changes won't happen.

Reply:
please help me understand where would the `access value` come from? because none of the accesstype methods would work if `setUser` fails.